src/threadwrappers.cpp

```diff
@@ -111,7 +111,7 @@ extern "C" int __clone(int (*fn) (void *arg), void *child_stack, int flags,
   if (tid == -1) {
     JTRACE("Clone call failed")(JASSERT_ERRNO);
     ThreadSync::decrementUninitializedThreadCount();
-    delete thread;
+    JALLOC_HELPER_FREE(thread);
```

Comment:
Why not change this to `ThreadList::threadIsDead(thread)`? That way we don't have to worry about ThreadList's internal memory management.